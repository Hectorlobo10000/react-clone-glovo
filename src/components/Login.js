/** @jsx jsx */
import React from "react";
import { jsx } from "@emotion/core";

import { Button, Card, Input } from "./ui";

function Login() {
  return (
    <div>
      <Card
        css={{
          maxWidth: "450px",
          position: "absolute",
          left: "50%",
          top: "50%",
          transform: "translate(-50%, -50%)"
        }}
      >
        <form>
          <h1
            css={{
              fontSize: "2rem",
              fontWeight: "normal",
              margin: "0"
            }}
          >
            Glovo
          </h1>

          <div
            css={{
              margin: "2rem 0"
            }}
          >
            <Input
              aria-label="Enter your e-mail"
              type="email"
              placeholder="Enter your e-mail"
              autoFocus
            />
          </div>

          <div
            css={{
              margin: "2rem 0"
            }}
          >
            <Input
              aria-label="Enter your password"
              type="password"
              placeholder="Enter your password"
              autoFocus
            />
          </div>

          <Button>Submit</Button>
        </form>
      </Card>
    </div>
  );
}

export default Login;
