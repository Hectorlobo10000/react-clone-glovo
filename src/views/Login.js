/** @jsx jsx */
import { jsx } from "@emotion/core";
import GlovoLogo from "../assets/glovo386.png";
import { navigate } from "@reach/router";
import { Button, Card, Input } from "../components/ui";
import { useUser } from "../selector";
import { useLogin } from "../action-hook";
import React from "react";

function Login() {
  const login = useLogin();
  const [email, setEmail] = React.useState("josh@delivery.pe");
  const [password, setPassword] = React.useState("123456");
  const user = useUser();

  React.useEffect(() => {
    if (user.name) {
      navigate("/");
    }
  }, [user]);

  function changeEmail(e) {
    setEmail(e.target.value);
  }

  function changePassword(e) {
    setPassword(e.target.value);
  }

  function handleSubmit(event) {
    event.preventDefault();
    login({ email, password });
  }

  return (
    <div>
      <Card
        css={{
          maxWidth: "450px",
          position: "absolute",
          left: "50%",
          top: "50%",
          transform: "translate(-50%, -50%)"
        }}
      >
        <form onSubmit={handleSubmit}>
          <img src={GlovoLogo} alt="Glovo Logo" />

          <div
            css={{
              margin: "2rem 0"
            }}
          >
            <Input
              aria-label="Enter your e-mail"
              type="email"
              placeholder="Enter your e-mail"
              autoFocus
              value={email}
              onChange={changeEmail}
            />
          </div>

          <div
            css={{
              margin: "2rem 0"
            }}
          >
            <Input
              aria-label="Enter your password"
              type="password"
              placeholder="Enter your password"
              autoFocus
              value={password}
              onChange={changePassword}
            />
          </div>

          <Button>Submit</Button>
        </form>
      </Card>
    </div>
  );
}

export default Login;
