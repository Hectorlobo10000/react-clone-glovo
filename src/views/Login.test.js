import React from "react";
import { render } from "@testing-library/react";
<<<<<<< HEAD

=======
>>>>>>> Check test components
import Login from "../views/Login";
test("Login Form on Submit", () => {
  const { asFragment } = render(<Login />);
  expect(asFragment()).toMatchSnapshot();
});
