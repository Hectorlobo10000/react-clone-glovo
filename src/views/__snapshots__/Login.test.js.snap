// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Form on Submit 1`] = `
<DocumentFragment>
  .emotion-5 {
  background: rgb(255,210,0);
  border-radius: .5em;
  box-shadow: 0px 2px 5px 0px rgba(0,0,0,.12);
  box-sizing: border-box;
  padding: 2rem;
  width: 100%;
  max-width: 450px;
  position: absolute;
  left: 50%;
  top: 50%;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.emotion-1 {
  margin: 2rem 0;
}

.emotion-0 {
  background: white;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  text-align: center;
  box-sizing: border-box;
  display: block;
  font-size: 1.5rem;
  padding: .5rem;
  width: 100%;
}

.emotion-0:focus {
  outline: none;
}

.emotion-4 {
  background-color: rgba(0,168,133,1);
  border: none;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: 1.2rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  font-weight: bold;
  text-transform: uppercase;
  width: 100%;
}

.emotion-4:hover {
  background-color: rgba(0,168,133,0.8);
  color: white;
}

<div>
    <div
      class="emotion-5"
    >
      <form>
        <img
          alt="Glovo Logo"
          src="glovo386.png"
        />
        <div
          class="emotion-1"
        >
          <input
            aria-label="Enter your e-mail"
            class="emotion-0"
            placeholder="Enter your e-mail"
            type="email"
          />
        </div>
        <div
          class="emotion-1"
        >
          <input
            aria-label="Enter your password"
            class="emotion-0"
            placeholder="Enter your password"
            type="password"
          />
        </div>
        <button
          class="emotion-4"
        >
          Submit
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
